src/main/java/com/_up/megastore/controllers/implementations/UserController.java

```diff
@@ -27,7 +28,10 @@ public void sendEmailToRecoverPassword(String email) {
   }
 
   @Override
-  public void recoverPassword(UUID userId, String newPassword, UUID recoverPasswordToken) {
-    userService.recoverPassword(userId, newPassword, recoverPasswordToken);
+  public void recoverPassword(UUID userId, RecoverPasswordRequest recoverPasswordRequest) {
+    userService.recoverPassword(userId,
+            recoverPasswordRequest.password(),
+            recoverPasswordRequest.confirmPassword(),
+            recoverPasswordRequest.recoverPasswordToken());
   }
```

Comment:
Con lo anterior, esto quedaría más conciso y dejamos al controlador limpio

```suggestion
  public void recoverPassword(UUID userId, RecoverPasswordRequest recoverPasswordRequest) {
    userService.recoverPassword(userId, recoverPasswordRequest);
  }
```